src/librustc_typeck/coherence/inherent.rs

```diff
@@ -295,12 +294,12 @@ impl<'a, 'tcx> InherentOverlapChecker<'a, 'tcx> {
 
             for &item2 in &impl_items2[..] {
                 if (name, namespace) == name_and_namespace(item2) {
-                    let msg = format!("duplicate definitions with name `{}`", name);
-                    let node_id = self.tcx.hir.as_local_node_id(item1).unwrap();
-                    self.tcx.sess.add_lint(lint::builtin::OVERLAPPING_INHERENT_IMPLS,
-                                           node_id,
-                                           self.tcx.span_of_impl(item1).unwrap(),
-                                           msg);
+                    struct_span_err!(self.tcx.sess,
+                                     self.tcx.span_of_impl(item1).unwrap(),
+                                     E0592,
+                                     "duplicate definitions with name `{}`",
+                                     name)
```

Comment:
The code looks good, but could you add a `span_label` to this diagnostic pointing at both `item1`'s and another pointing at `item2`'s span?

That way the code output would look this way:

```rust
error: duplicate definitions with name `id`
 --> <anon>:4:5
  |
4 |     fn id() {} //~ ERROR duplicate definitions
  |     ^^^^^^^^^^ duplicate definitions for `id`
...
8 |     fn id() {}
  |     ---------- other definition for `id`
```